pkg/controller/backup_controller.go

```diff
@@ -382,7 +384,12 @@ func (controller *backupController) runBackup(backup *api.Backup, bucket string)
 		return err
 	}
 
-	objectStore, err := getObjectStore(controller.objectStoreConfig, pluginManager)
+	backupLocation, err := controller.backupLocationLister.BackupStorageLocations(backup.Namespace).Get(backup.Spec.StorageLocation)
```

Comment:
I would just pass `backupLocation` into this func in place of `bucket`, then you won't have to fetch it again